.github/workflows/shellcheck.yml

```diff
@@ -0,0 +1,65 @@
+name: Shellcheck Lint
+
+on:
+  push:
+    paths:
+      # Run workflow on every push
+      # only if a file within the specified paths has been changed:
+      - 'tests/**'
+      - 'usr/**'
+
+  pull_request:
+    paths:
+      # Run workflow on every push
+      # only if a file within the specified paths has been changed:
+      - 'tests/**'
+      - 'usr/**'
+
+  # Allows you to run this workflow manually from the Actions tab
+  workflow_dispatch:
+
+jobs:
+  build:
+    name: Shellcheck Lint
+
+    # This job runs on Linux
+    runs-on: ubuntu-latest
+
+    steps:
+      # Required to access files of this repository
+      - uses: actions/checkout@v2
+
+      # Download Shellcheck and add it to the workflow path
+      - name: Download Shellcheck
+        run: |
+          wget -qO- "https://github.com/koalaman/shellcheck/releases/download/stable/shellcheck-stable.linux.x86_64.tar.xz" | tar -xJv
+          chmod +x shellcheck-stable/shellcheck
+      # Verify that Shellcheck can be executed
+      - name: Check Shellcheck Version
+        run: |
+          shellcheck-stable/shellcheck --version
+
+      # Run Shellcheck on repository
+      # ---
+      # https://github.com/koalaman/shellcheck
+      # ---
+      # Excluded checks:
+      # https://www.shellcheck.net/wiki/SC1091 -- Not following: /etc/rc.status was...
+      # https://www.shellcheck.net/wiki/SC1090 -- Can't follow non-constant source. ..
+      # ---
+      - name: Run Shellcheck
```

Comment:
I can't follow your thoughts… 🤔 What exactly is "too much code in CI definitions" for you?
You can run [lines 52-64](https://github.com/rear/rear/pull/2756/files/3e16eee9b71895a31ac8433f577f6a59ff67bf00#diff-549859aac5bcaebdf605d3ffd525377f2b314b5cc6eb4ebad490bd2cc52617a7R52-R64) very well "on your laptop" (if you have shellcheck installed or running it in a local docker container).

See it run this "on my computer":
```
🦎🖥  ✔ ~/temp/PRs/rear [issue_1040_shellchecked_version|✚ 2…8]
23:49 $ bash 1
shellcheck'ing ./tests/setup1/run.sh
shellcheck'ing ./tests/setup1/tcase1.sh
shellcheck'ing ./tests/setup1/tcase2.sh
shellcheck'ing ./tests/setup1/tcase3.sh
shellcheck'ing ./tests/setup1/tcase4.sh
shellcheck'ing ./tests/setup1/tcase5.sh
shellcheck'ing ./tests/setup1/tcase6.sh
shellcheck'ing ./tests/setup1/tcase7.sh
shellcheck'ing ./tests/setup1/tcase8.sh
shellcheck'ing ./tests/setup1/verify.sh
shellcheck'ing ./tests/setup2/tcase1.sh
shellcheck'ing ./tests/setup2/tcase2.sh
shellcheck'ing ./tests/setup2/tcase3.sh
shellcheck'ing ./tests/setup2/tcase4.sh
shellcheck'ing ./tests/setup2/tcase5.sh
shellcheck'ing ./tests/setup2/tcase6.sh
shellcheck'ing ./tests/setup2/tcase7.sh
shellcheck'ing ./tests/setup2/tcase8.sh
shellcheck'ing ./usr/sbin/rear
shellcheck'ing ./usr/share/rear/lib/write-protect-functions.sh
shellcheck'ing ./usr/share/rear/skel/default/bin/dhclient-script
shellcheck'ing ./usr/share/rear/skel/default/bin/dhcpcd.sh
shellcheck'ing ./usr/share/rear/skel/default/bin/ifup
shellcheck'ing ./usr/share/rear/skel/default/bin/login
shellcheck'ing ./usr/share/rear/skel/default/etc/scripts/boot
shellcheck'ing ./usr/share/rear/skel/default/etc/scripts/do-shutdown
shellcheck'ing ./usr/share/rear/skel/default/etc/scripts/run-serial
shellcheck'ing ./usr/share/rear/skel/default/etc/scripts/run-sshd
shellcheck'ing ./usr/share/rear/skel/default/etc/scripts/run-syslog
shellcheck'ing ./usr/share/rear/skel/default/etc/scripts/system-setup
shellcheck'ing ./usr/share/rear/skel/default/etc/scripts/system-status.sh
shellcheck'ing ./usr/share/rear/skel/default/etc/scripts/unlock-opal-disks
Shellcheck failed for one or more shellscript(s)
🦎🖥  ✔ ~/temp/PRs/rear [issue_1040_shellchecked_version|✚ 2…8]
23:50 $ cat 1
          set +e
          find  ./*/ -type f | sort | while read -r sh; do
            if [ "$(file --brief --mime-type "$sh")" == 'text/x-shellscript' ]; then
              echo "shellcheck'ing $sh"
              if ! docker run --rm -v "$PWD:/mnt" koalaman/shellcheck:stable -Calways --color=always --severity=warning --exclude=SC1091,SC1090 "$sh"; then
                touch some_scripts_have_failed_shellcheck
              fi
            fi
          done
          if [ -f ./some_scripts_have_failed_shellcheck ]; then
            echo "Shellcheck failed for one or more shellscript(s)"
            exit 1
          fi
🦎🖥  ✔ ~/temp/PRs/rear [issue_1040_shellchecked_version|✚ 2…8]
23:51 $
```

Reply:
Looking at your example you actually had to copy parts of the [.github/workflows/shellcheck.yml](https://github.com/rear/rear/pull/2756/files/3e16eee9b71895a31ac8433f577f6a59ff67bf00#diff-549859aac5bcaebdf605d3ffd525377f2b314b5cc6eb4ebad490bd2cc52617a7) GitHub workflow into a file `1` and then run this file. For me this is an unnecessary step and also error prone.

Instead, add this code to a script or directly to the [Makefile](https://github.com/rear/rear/blob/master/Makefile) so that `make validate` will also run these checks (and maybe we would want to rename `make validate` to `make test` as that name is more common).

The goal is DRY (don't repeat yourself) and to make it simple to run CI tests locally without any extra effort.

Reply:
I agree to add this to your `Makefile` - but if `shellcheck` is not installed, there's no easy way to install it on each different linux distro than you or me might use (openSUSE). So we make an error message and exit with RC=1. 👍🏻

Reply:
Don't worry about installing `shellcheck`, it's OK to fail in the `Makefile` when dependencies are missing, we do this with the other stuff as well. It is in general not the job of a `Makefile` to install all build requirements but it can assume that they are usable.

Reply:
I even think it is basically forbidden for common security reasons
that running something (as root) automatically installs
additional (external) stuff onto the current system
without explicit prior confirmation from the user
except it is clear to the user in advance that running something
is meant to and will install (external) things on his system.

In this case running "make" is not expected to install something.
Even running "make install" is not expected to install external stuff.

I know - I know - nowadays things easily just work simply via
```
curl $URL | sudo sh
```